qml/MainPage.qml

```diff
@@ -246,18 +246,8 @@ Item {
                 print(FileManager.createFileName(FileManager.Pictures))
                 result.saveToFile(FileManager.createFileName(FileManager.Pictures))
             })
-        } else if (event.key == Qt.Key_W) {
-            print("Grab waterfall image")
-            event.accepted = true
-
-            ping1DVisualizer.waterfallItem.grabToImage(function(result) {
-                print("Grab waterfall image callback")
-                print(FileManager.createFileName(FileManager.Pictures))
-                result.saveToFile(FileManager.createFileName(FileManager.Pictures))
-            })
-        } else if (event.key == Qt.Key_R) {
-            print("Clear waterfall")
-            ping1DVisualizer.waterfallItem.clear()
+        } else {
```

Comment:
I think this might be a move in the wrong direction.

It seems not the job of the visualizer to handle keyboard keys. An idea instead of doing the keyboard handling for each visualizer, maybe add grabtoimage and clear functions to the visualizer, and call those with the keyboard shortcut from main qml here.

Reply:
Fixed.